modules/server/templates/scripts/forseti-server/forseti_server_startup_script.sh.tpl

```diff
@@ -4,45 +4,57 @@ set -eu
 # Env variables
 USER=ubuntu
 USER_HOME=/home/ubuntu
+INTERNET_CONNECTION="$(ping -q -w1 -c1 google.com &>/dev/null && echo online || echo offline)"
+
+# Log status of internet connection
+if [ $INTERNET_CONNECTION == "offline" ]; then
+  echo "Forseti Startup - A connection to the internet was not detected."
+fi
 
 # forseti_conf_server digest: ${forseti_conf_server_checksum}
 # This digest is included in the startup script to rebuild the Forseti server VM
 # whenever the server configuration changes.
 
 # Ubuntu update.
+echo "Forseti Startup - Updating Ubuntu."
 sudo apt-get update -y
 sudo DEBIAN_FRONTEND=noninteractive apt-get -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" upgrade
 sudo apt-get update -y
 sudo apt-get --assume-yes install google-cloud-sdk git unzip
 
 if ! [ -e "/usr/sbin/google-fluentd" ]; then
-    cd $USER_HOME
-    curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
-    bash install-logging-agent.sh
+  echo "Forseti Startup - Installing GCP Logging agent."
+  cd $USER_HOME
+  curl -sSO https://dl.google.com/cloudagents/install-logging-agent.sh
+  bash install-logging-agent.sh
 fi
 
 # Check whether Cloud SQL proxy is installed.
 if [ -z "$(which cloud_sql_proxy)" ]; then
-      cd $USER_HOME
-      wget https://dl.google.com/cloudsql/cloud_sql_proxy.${cloudsql_proxy_arch}
-      sudo mv cloud_sql_proxy.${cloudsql_proxy_arch} /usr/local/bin/cloud_sql_proxy
-      chmod +x /usr/local/bin/cloud_sql_proxy
+  echo "Forseti Startup - Installing GCP Cloud SQL Proxy."
+  cd $USER_HOME
+  wget https://dl.google.com/cloudsql/cloud_sql_proxy.${cloudsql_proxy_arch}
+  sudo mv cloud_sql_proxy.${cloudsql_proxy_arch} /usr/local/bin/cloud_sql_proxy
+  chmod +x /usr/local/bin/cloud_sql_proxy
 fi
 
 # Install Forseti Security.
 cd $USER_HOME
-rm -rf *forseti*
+if [ $INTERNET_CONNECTION == "online" ]; then
+  rm -rf *forseti*
+fi
```

Comment:
I added this check in to not remove the Forseti installation if there is no public internet connection, as was reported from [this Forseti issue](https://github.com/forseti-security/forseti-security/issues/3186#issue-491757730).